src/basic-samples/GuaranteedSubscriber.js

```diff
@@ -0,0 +1,254 @@
+
+    // main function
+    subscriber.run = function (argv) {
+        subscriber.connect(argv);
+    };
+
+    // Establishes connection to Solace message router
+    subscriber.connect = function (argv) {
+        if (subscriber.session !== null) {
+            subscriber.log('Already connected and ready to consume messages.');
+            return;
+        }
+
+        // extract params
+        if (argv.length < (2 + 3)) { // expecting 3 real arguments
+            subscriber.log('Cannot connect: expecting all arguments' +
+                ' <protocol://host[:port]> <client-username>@<message-vpn> <client-password>.\n' +
+                'Available protocols are ws://, wss://, http://, https://, tcp://, tcps://');
+            process.exit();
+        }
+        var hosturl = argv.slice(2)[0];
+        subscriber.log('Connecting to Solace message router using url: ' + hosturl);
+        var usernamevpn = argv.slice(3)[0];
+        var username = usernamevpn.split('@')[0];
+        subscriber.log('Client username: ' + username);
+        var vpn = usernamevpn.split('@')[1];
+        subscriber.log('Solace message router VPN name: ' + vpn);
+        var pass = argv.slice(4)[0];
+
+        // create session
+        try {
+            subscriber.session = solace.SolclientFactory.createSession({
+                // solace.SessionProperties
+                url:      hosturl,
+                vpnName:  vpn,
+                userName: username,
+                password: pass,
+            });
+        } catch (error) {
+            subscriber.log(error.toString());
+        }
+
+        // define session event listeners
+        subscriber.session.on(solace.SessionEventCode.UP_NOTICE, function (sessionEvent) {
+            subscriber.log('=== Successfully connected and ready to start the message subscriber. ===');
+            subscriber.startConsume();
+        });
+        subscriber.session.on(solace.SessionEventCode.CONNECT_FAILED_ERROR, function (sessionEvent) {
+            subscriber.log('Connection failed to the message router: ' + sessionEvent.infoStr +
+                ' - check correct parameter values and connectivity!');
+        });
+        subscriber.session.on(solace.SessionEventCode.DISCONNECTED, function (sessionEvent) {
+            subscriber.log('Disconnected.');
+            subscriber.consuming = false;
+            if (subscriber.session !== null) {
+                subscriber.session.dispose();
+                subscriber.session = null;
+            }
+        });
+
+        subscriber.connectToSolace();   
+    };
+
+    // Actually connects the session triggered when the iframe has been loaded - see in html code
+    subscriber.connectToSolace = function () {
+        try {
+            subscriber.session.connect();
+        } catch (error) {
+            subscriber.log(error.toString());
+        }
+    };
+
+    // Starts consuming from a queue on Solace message router
```

Comment:
Can we please change the comment to `...on Solace PubSub+ Event Broker`

Reply:
Looks like there are multiple references to Solace message router, I would recommend changing all those as well while you're at it...

Reply:
Good point. Almost all samples use the term "Solace message router".  It would look awkward to fix in one file, I am taking the liberty to fix ALL the files.